client/shipping-label/state/reducer.js

```diff
@@ -140,7 +150,235 @@ reducers[ UPDATE_PACKAGE_WEIGHT ] = ( state, { packageId, value } ) => {
+			openedPackageId = targetPackageId;
+		}
+	}
+
+	return {
+		...state,
+		movedItemIndex: -1,
+		showItemMoveDialog: false,
+		openedPackageId,
+		form: {
+			...state.form,
+			packages: {
+				...state.form.packages,
+				selected: newPackages,
+				unpacked: newUnpacked,
+				saved: false,
+			},
+			rates: {
+				...state.form.rates,
+				values: _.mapValues( newPackages, ( pckg ) => ( pckg.service || '' ) ),
+			},
+		},
+	};
+};
+
+reducers[ CLOSE_ITEM_MOVE ] = ( state ) => {
+	return {
+		...state,
+		movedItemIndex: -1,
+		showItemMoveDialog: false,
+	};
+};
+
+reducers[ SET_TARGET_PACKAGE ] = ( state, { targetPackageId } ) => {
+	return {
+		...state,
+		targetPackageId,
+	};
+};
+
+reducers[ ADD_PACKAGE ] = ( state ) => {
+	const newPackages = {...state.form.packages.selected};
+	const packageKeys = Object.keys( newPackages );
+	const boxesKeys = Object.keys( state.form.packages.all );
+	if ( ! boxesKeys.length ) {
+		return state;
+	}
+
+	const boxId = boxesKeys[ 0 ];
+	const box = state.form.packages.all[ boxId ];
+	const { height, length, width } = getBoxDimensions( box );
+
+	const keyBase = 'client_custom_';
+	let openedPackageId;
+	for ( let i = 0; i <= packageKeys.length; i++ ) {
+		if ( -1 === packageKeys.indexOf( keyBase + i ) ) {
+			newPackages[ keyBase + i ] = {
+				height, length, width,
+				id: keyBase + i,
+				weight: box.box_weight,
+				box_id: boxId,
+				items: [],
+			};
+
+			openedPackageId = keyBase + i;
+
+			break;
+		}
+	}
+
+	return {
+		...state,
+		openedPackageId,
+		form: {
+			...state.form,
+			packages: {
+				...state.form.packages,
+				selected: newPackages,
+				saved: false,
+			},
+			rates: {
+				...state.form.rates,
+				values: _.mapValues( newPackages, ( newPckg ) => ( newPckg.service || '' ) ),
+			},
+		},
+	};
+};
+
+reducers[ REMOVE_PACKAGE ] = ( state, { packageId } ) => {
+	const newPackages = {...state.form.packages.selected};
+	const pckg = newPackages[ packageId ];
+	const newUnpacked = state.form.packages.unpacked.concat( pckg.items );
+	delete newPackages[ packageId ];
+	const openedPackageId = Object.keys( newPackages )[ 0 ] || '';
+
+	return {
+		...state,
+		openedPackageId,
+		form: {
+			...state.form,
+			packages: {
+				...state.form.packages,
+				selected: newPackages,
+				unpacked: newUnpacked,
+				saved: false,
+			},
+			rates: {
+				...state.form.rates,
+				values: _.mapValues( newPackages, ( newPckg ) => ( newPckg.service || '' ) ),
+			},
+		},
+	};
+};
+
+reducers[ SET_PACKAGE_TYPE ] = ( state, { packageId, boxTypeId } ) => {
+	const box = state.form.packages.all[ boxTypeId ];
+	const newPackages = {...state.form.packages.selected};
+	const { length, width, height } = getBoxDimensions( box );
+	const oldPackage = newPackages[ packageId ];
+
+	newPackages[ packageId ] = {
+		...oldPackage,
+		height, length, width,
+		weight: box.box_weight + oldPackage.items.reduce( ( sum, item ) => ( sum + item.weight ), 0 ),
```

Comment:
Consider using lodash's [`sumBy()`](https://lodash.com/docs/4.16.4#sumBy) for this.